lib/API/JSON/v1/Settings.php

```diff
@@ -51,6 +57,16 @@ function set($settings = []) {
       ) {
         $this->onInactiveSubscribersIntervalChange();
       }
+
+      // Sync WooCommerce Customers list on settings change
+      $new_subscribe_old_woocommerce_customers = 0;
+      if (isset($settings['mailpoet_subscribe_old_woocommerce_customers']['enabled'])) {
+        $new_subscribe_old_woocommerce_customers = (int)$settings['mailpoet_subscribe_old_woocommerce_customers']['enabled'];
+      }
+      if ($original_subscribe_old_woocommerce_customers !== $new_subscribe_old_woocommerce_customers) {
+        $this->woocommerce_segment->synchronizeCustomers();
```

Comment:
We should schedule this task and let the cron run it. This can take a long time to run.